src/main/profiling_info.cpp

```diff
@@ -42,41 +66,66 @@ bool ProfilingInfo::Enabled(const MetricsType setting) const {
 	if (setting == MetricsType::OPERATOR_TIMING && Enabled(MetricsType::CPU_TIME)) {
 		return true;
 	}
+	if (setting == MetricsType::OPERATOR_CARDINALITY && Enabled(MetricsType::CUMULATIVE_CARDINALITY)) {
+		return true;
+	}
 	return false;
 }
 
+string ProfilingInfo::GetMetricAsString(MetricsType setting) {
+	if (!Enabled(setting)) {
+		throw InternalException("Metric %s not enabled", EnumUtil::ToString(setting));
+	}
+
+	if (setting == MetricsType::EXTRA_INFO) {
+		string result;
+		for (auto &it : extra_info) {
+			if (!result.empty()) {
+				result += ", ";
+			}
+			result += StringUtil::Format("%s: %s", it.first, it.second);
+		}
+		return "\"" + result + "\"";
+	}
+
+	if (metrics[setting].IsNull()) {
+		return "0";
+	}
```

Comment:
When would an enabled metric be `NULL`? Can this happen, or should this also be an `InternalException`?

Reply:
Good point. I actually added this in to catch a bug but fixed it earlier, so I think this can be a `D_ASSERT` instead since it shouldn't be `NULL` here.